### PR TITLE
refactor: insert select to stream mode

### LINKED ISSUE #1542: INSERT INTO SELECT in a streaming way
### Describe This Problem

Followup for https://github.com/apache/horaedb/pull/1536, now we will collect all rows to vec before do insert, it's best if we can do insert select in a streaming way.

### Proposal

Support insert select like
```rs
while let Some(rows) = query.next_batch().await? {
  insert(rows);
}
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by zealchen**
## Rationale
Close #1542 

## Detailed Changes
Do select and insert procedure in stream way.

## Test Plan
CI test.

### CODE REVIEW

File: src/interpreters/src/insert.rs [hunk 1]
```diff
@@ -168,38 +179,98 @@ impl Interpreter for InsertInterpreter {
                 .await
                 .context(Insert)?;
 
-                if record_batches.is_empty() {
-                    return Ok(Output::AffectedRows(0));
+                // accumulate 8 record batches before writing to table and this param
+                // can be adjusted according to the performance test result.
+                let max_batch_size = 8;
```
Comment (jiacai2050) [inline src/interpreters/src/insert.rs:184]: I think `max_row_count` is more suitable here, 1000 is a reasonable default value.

File: src/interpreters/src/insert.rs [hunk 2]
```diff
@@ -168,38 +179,98 @@ impl Interpreter for InsertInterpreter {
...
+                let mut result_rows = 0;
+                let mut record_batches = Vec::new();
+                while let Some(response) = record_batches_stream.next().await {
```
Comment (jiacai2050) [inline src/interpreters/src/insert.rs:187]:
```rust
while let Some(response) = record_batches_stream.try_next().await? {
  ...
}
```

This looks more elegant.

**Comment by zealchen** [comment]
Refactor to do the select and insert in parallel way.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**